spec/template.dd

```diff
@@ -637,6 +642,44 @@ $(GNAME TemplateAliasParameterDefault):
         }
         ------
         )
+
+        $(LI Compile-time values
+
+        ------
+        template Foo(alias x)
+        {
+            static int i = x;
+        }
+
+        void main()
+        {
+            // compile-time argument evaluation
+            enum two = 1 + 1;
+            alias foo = Foo!(5 * two);
+            assert(foo.i == 10);
+            static assert(foo.stringof == "Foo!10");
+
+            // compile-time function evaluation
+            int get10() { return 10; }
+            alias bar = Foo!(get10());
+            // bar is the same template instance as foo
+            assert(&bar.i is &foo.i);
+        }
+        ------
+        )
+
+        $(LI Lambdas
+
+        ------
+        template Foo(alias fun)
+        {
+            enum val = fun(2);
+        }
+
+        alias foo = Foo!(x => x * x);
```

Comment:
`x => x * x` is not a value. It has no type. It's a template. `(int x) => x * x` would be a value.